SDLMTCloud.Provider/Sdl.Community.MTCloud.Provider/Studio/SdlMTCloudTranslationProvider.cs

```diff
@@ -35,12 +36,38 @@ public class SdlMTCloudTranslationProvider : ITranslationProvider
 			TranslationService = translationService;
 			_editorController = editorController;
 			_firstTimeAdded = firstTimeAdded;
+			_rateItController = SdlTradosStudio.Application.GetController<RateItController>();
+
+			var projectsController = MtCloudApplicationInitializer.ProjectsController;
+			projectsController.CurrentProjectChanged -= ProjectsController_CurrentProjectChanged;
+			projectsController.CurrentProjectChanged += ProjectsController_CurrentProjectChanged;
+			_currentProject = projectsController.CurrentProject.GetProjectInfo();
 
 			LoadState(translationProviderState);
+			ActivateRatingController();
 
 			MtCloudApplicationInitializer.Subscribe<TranslationProviderStatusChanged>(Settings_TranslationProviderStatusChanged);
 		}
 
+		private void ProjectsController_CurrentProjectChanged(object sender, EventArgs e)
+		{
+			if (sender is not ProjectsController projController) return;
+			var newProject = projController.CurrentProject.GetProjectInfo();
+			if (newProject == _currentProject) return;
+			_currentProject = newProject;
+
+			var tpState =
+				projController.CurrentProject.GetTranslationProviderConfiguration().Entries.FirstOrDefault(
+					entry => entry.MainTranslationProvider.Uri.ToString().Contains("sdlmtcloud"))?.MainTranslationProvider.State;
+
+			var currentLanguagePair = new LanguagePair(_currentProject.SourceLanguage.CultureInfo,
+				_editorController?.ActiveDocument?.ActiveFile.Language.CultureInfo);
+
+				GetMTCloudLanguagePair(currentLanguagePair);
+			LoadState(tpState);
```

Comment:
Do we check for null in another place of the application? tpState can be null

Reply:
the commit you were looking at was a bit behind... the tpState is checked for null